sim/side.ts

```diff
@@ -314,19 +314,16 @@ export class Side {
 		this.battle.send('sideupdate', `${this.id}\n${sideUpdate}`);
 	}
 
-	emitCallback(...args: (string | number | AnyObject)[]) {
-		this.battle.send('sideupdate', `${this.id}\n|callback|${args.join('|')}`);
-	}
-
 	emitRequest(update: AnyObject) {
 		this.battle.send('sideupdate', `${this.id}\n|request|${JSON.stringify(update)}`);
 		this.activeRequest = update;
 	}
 
-	emitChoiceError(message: string) {
+	emitChoiceError(message: string, unavailable?: boolean) {
 		this.choice.error = message;
-		this.battle.send('sideupdate', `${this.id}\n|error|[Invalid choice] ${message}`);
-		if (this.battle.strictChoices) throw new Error(`[Invalid choice] ${message}`);
+		const type = `[${unavailable ? 'Unavailable' : 'Invalid'} choice]`;
```

Comment:
Bikeshedding time! What should be the new name for "Unavailable choice"?

I'm currently leaning towards "Invalid choice".

Reply:
Oh god, I should have merged it myself after you approved ;)

We can't make the name "Invalid choice", the whole point is that its *different* from "Invalid choice" so that it can be distinguished. Do you mean "Disabled choice"? I'm indifferent

Reply:
Err, I mean "Disabled choice". Typo on my part.

Reply:
"Unavailable" mainly strikes me as kind of complicated for English. It might not play as well with non-native speakers, even if it is more precise/accurate than "Disabled".

Reply:
Redundancy is maybe a concern. "[Unavailable] Move x is disabled" is less redundant than "[Disabled] Move x is disabled".

Other question: "[Unavailable]" vs "[Unavailable choice]"?

Reply:
(None of this actually matters; feel free to merge.)

Reply:
I went with "Unavailable choice" as opposed to "Disabled choice" for the redundancy/stutter it introduces, as you identified. I also think "choice" is useful in the description for symmetry with "Invalid choice" and because "Unavailable" makes it seem like simulator itself is unavailable (like a 500 error on a web server) instead of the choice that was made being unavailable.

I'm also not in love with "Unavailable choice" FWIW, it just seems like the least bad option.

Reply:
Fair. I have nothing to add there.